### PR TITLE
Add more CA usage best practices

### PR DESCRIPTION
**Pull request by cipherboy**
First pass at some additional CA best practices.

Still missing:

 - [x] ACLing of paths by persona

### CODE REVIEW

File: website/content/docs/secrets/pki/considerations.mdx [hunk 1]
```diff
@@ -217,6 +297,71 @@ nature:
...
+   revocation on behalf of an operator; may also handle delegated
+   issuance.
+ - *Advanced*; potentially a power-user that has access to additional
+   issuance APIs.
+ - *Requester*; a low-level user that simply requests certificates.
+ - *Unauthed*; any arbitrary user or service that lacks a Vault token.
+
+For these personas, we suggest the following ACLs, in condensed, tabular form:
+
+| Path | Operations | Operator | Agent | Advanced | Requester | Unauthed |
```
Comment (cipherboy) [inline website/content/docs/secrets/pki/considerations.mdx:323]: @ivana-mcconnell made a comment out of band that these might be good to have per-persona; currently they're sorted as such (so all the persona's paths are grouped together and you can then read across the table to find exactly those paths), but I'm not sure if it makes sense expanding these per-persona explicitly (duplicating endpoints across various personas). Does anyone know if we can do those fancy tables like on the tutorials pages (which I don't think are hosted on this repo)? 

This present table makes the additive nature visually clear, but sadly it is already too long to fit on a single (horizontal) screen, so maybe it doesn't matter and breaking it up would be more useful in general.

Should we make a better effort to make these in an ACL-able format (i.e., remove the regex patterns)? Or should we explicitly leave them in regex-style paths, to force people to think about the ACL actions they're potentially choosing here, as they are in general, risky?

Thoughts?

Comment (stevendpclark) [reply website/content/docs/secrets/pki/considerations.mdx:324]: This is really nice, but as you say quite long sadly. It's even missing some paths :( Pre-canned policies that enforce these personas might be interesting?

issuer/:issuer_ref/sign-intermediate
issuer/:issuer_ref/sign-self-issued
root/rotate/:exported

Comment (cipherboy) [reply website/content/docs/secrets/pki/considerations.mdx:324]: The concern with pre-canned policies in particular is that you might rarely want to actually open up _all_ users to issue against _all_ roles; I think more securely, we'd recommend limiting users to specific roles, which is up to the operator to design and implement.

That alone makes the pre-canned approach harder, as we're just doing the "easy" (operator/agent) side and leaving the (advanced/requester) roles more broad.

Comment (stevendpclark) [reply website/content/docs/secrets/pki/considerations.mdx:324]: Fair, but we could have a pre-canned Requester policy with <:role_name:> or something in that regard to push them down that path?

**Review by stevendpclark — APPROVED** [review]
💯

**Pull request merged**